src/nbl/ext/MitsubaLoader/CMitsubaMaterialCompilerFrontend.cpp

```diff
@@ -86,27 +87,59 @@ CMitsubaMaterialCompilerFrontend::EmitterNode* CMitsubaMaterialCompilerFrontend:
     if (_emitter->area.emissionProfile) {
         asset::material_compiler::IR::CEmitterNode::EmissionProfile profile;
         auto [image, sampler, meta] = getEmissionProfile(_emitter->area.emissionProfile);
-        if (image) {
-            profile.texture = { image, sampler, 1.f };
-            auto inverseTransform = core::concatenateBFollowedByA(transform, _emitter->transform.matrix);
-            inverseTransform[1].w = 0;
-            inverseTransform[2].w = 0;
-            profile.right_hand = core::determinant(inverseTransform) >= 0.0f;
-            profile.up = core::normalize(inverseTransform[1]);
-            profile.view = core::normalize(inverseTransform[2]);
+        
+        auto fillProfile = [&]() -> void
+        {
+            if (_emitter->area.emissionProfile->flatten > 0)
+            {
+                auto clone = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(image->clone()); // TODO: check if its already a copy
+                if (!asset::CIESProfile::flattenIESTexture(meta->profile.getAvgEmmision(), clone, _emitter->area.emissionProfile->flatten))
+                {
+                    os::Printer::log("ERROR: Emission profile rejected because flatten operation on the IES texture failed!", ELL_ERROR);
+                    return; // exit the lambda
+                }
+;
+                profile.texture = { clone, sampler, 1.f };
+            }
+            else
+                profile.texture = { image, sampler, 1.f };
 
-            res->emissionProfile = profile;
-            float normalizeEnergy = _emitter->area.emissionProfile->normalizeEnergy;
-            if (normalizeEnergy == 0.0f) {
-                res->intensity *= meta->getMaxIntensity();
+
+            auto worldSpaceIESTransform = core::concatenateBFollowedByA(transform, _emitter->transform.matrix);
+            
+            // fill .w component of 1 & 2 row with 0 to perform normmalization on .xyz vectors for these rows bellow to get up & view vectors
+            worldSpaceIESTransform[1].w = 0;
+            worldSpaceIESTransform[2].w = 0;
+
+            _NBL_STATIC_INLINE_CONSTEXPR float THRESHOLD = std::numeric_limits<float>::epsilon();
```

Comment:
use a higher threshold like `exp2(-14.f)`

Reply:
resolved in following commit